src/Network/Http.php

```diff
@@ -520,7 +520,22 @@ public function setOption($option, $value = null)
             return;
         }
 
-        $this->requestOptions[$option] = $value;
+        if (is_string($option) && defined($option)) {
+            $optionKey = constant($option);
+            $this->requestOptions[$optionKey] = $value;
+        } elseif (is_int($option)) {
+            $constants = get_defined_constants(true);
+            $curlOptConstants = preg_grep('/^CURLOPT_/', array_flip($constants['curl']));
+
+            if (isset($curlOptConstants[$option])) {
+                $this->requestOptions[$option] = $value;
+            } else {
+                throw new \Exception('Array keys must be CURLOPT constants or equivalent integer values');
```

Comment:
@mahony0 Would you mind changing the Exception type to `October\Rain\Exception\ApplicationException`, to reflect that it is an implementation error by the developer?

Reply:
In addition, would you also mind changing the message to the following?
```suggestion
                throw new \Exception('$option parameter must be a CURLOPT constant or equivalent integer');
```